spec/generator/gen-named-callout.spec.js

```diff
@@ -0,0 +1,107 @@
+import {toOnigurumaAst} from '../../dist/index.js';
+import {generate} from '../../dist/generator/generate.js';
+
+describe('generate: Named Callout', () => {
+  function gen(pattern) {
+    return generate(toOnigurumaAst(pattern)).pattern;
+  }
+
+  it('should support all named callout names', () => {
+    const cases = [
+      ['(*FAIL)'],
+      ['(*MISMATCH)'],
+      ['(*SKIP)'],
+      ['(*ERROR)'],
+      ['(*MAX{1})'],
+      ['(*COUNT)'],
+      ['(*TOTAL_COUNT)'],
+      ['(*CMP{5,>=,4})'],
+      // ['(*foo)'], // Error: Invalid callout name "foo"
```

Comment:
IMO better to test invalid names in a separate `it` function, and validate that they throw. (Look for existing cases of `toThrow` in the code to see an example.)